zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/LegacyKeyringImpl.java

```diff
@@ -306,6 +309,35 @@ public void unlock(User user, String password) throws KeyringException {
         }
     }
 
+    @Override
+    public void populate(User src, User target, Set<String> collectionIDs) throws KeyringException {
+        validateUser(src);
+
+        com.github.onsdigital.zebedee.json.Keyring srcKeyring = getCachedUserKeyring(src);
+        if (srcKeyring == null) {
+            throw new KeyringException(CACHE_KEYRING_NULL_ERR);
+        }
+
+        if (!srcKeyring.isUnlocked()) {
+            throw new KeyringException(KEYRING_LOCKED_ERR);
+        }
+
+        validateUser(target);
+
+        if (collectionIDs == null || collectionIDs.isEmpty()) {
```

Comment:
Do we want to check this before getting the cachedUserKeyring?

Reply:
Some context might help explain this a bit more clearly:

When we create a new user their keyring needs to be populate with the keys for the collection they should have access too. In the current world this is done by **reading the key values from the user who created** the new user.

You can write a new key to a keyring at anytime (even if its locked) but you can only read from a keyring that has been unlocked. When a user logs in their keyring is unlocked and added to the cache - this enables us to read keys from it later on without requiring their password each time.

In the context of this functionality if the keyring of the user who is is creating a new user is not in the cache then something has gone wrong and we cannot read any values from it. Even if we don't have any keys to add this is scenario would be the symptom of something having gone very wrong which is why we should check it. This of it as a mandatory prerequisite and it's not met we need to sound the alarm

If everything is as it should be and there are no keys to add then we essentially don't need to do anything. However, I think I could move this check above `validateTargetUser()` as it would be more efficient i.e. only valid this target user if we know we need to add keys.